src/coreclr/vm/object.h

```diff
@@ -1466,6 +1469,9 @@ class AssemblyLoadContextBaseObject : public Object
     CLR_BOOL      _isCollectible;
 #else // TARGET_64BIT
     int64_t       _id; // On 32-bit platforms this 64-bit value type is larger than a pointer so JIT places it first
+    OBJECTREF     _appContext;
+    OBJECTREF     _assemblySimpleNameMvidCheckHash;
+    OBJECTREF     _loadedAssemblies;
```

Comment:
There's layout mismatch asserted only on ARM32. Need some help about field layout since I don't have ARM environment.

Reply:
([wild guess](https://github.com/dotnet/coreclr/pull/23906/files)) Does it help if we enable `pshpack4` and `poppack` (from line 1447 and 1493) for unix arm? There is currently a four byte difference between managed and unmanaged object sizes.